### ISSUE TITLE
Mighty Switch Force! Collection (832320)

### ISSUE THREAD

**Issue by AlleyDismay** [ValveSoftware/Proton#7062]
# Compatibility Report
- Name of the game with compatibility issues: Mighty Switch Force! Collection
- Steam AppID of the game: 832320

## System Information
https://gist.github.com/AlleyDismay/37f37f8613267d042d38cbbac09f2098

## I confirm:
- [X] that I haven't found an existing compatibility report for this game.
- [X] that I have checked whether there are updates for my system available.

[steam-832320 - Proton 8.0-3.log](https://github.com/ValveSoftware/Proton/files/12489531/steam-832320.-.Proton.8.0-3.log)
[steam-832320 - Proton 7.0-6.log](https://github.com/ValveSoftware/Proton/files/12489532/steam-832320.-.Proton.7.0-6.log)

## Symptoms
Get into game selection screen and use arrow keys to scroll up and down the games. Getting into a game fast enough may prevent the crash.

## Known info:
https://www.protondb.com/app/832320 - Says it's Platinum. No issue reports have been made that I can find.

## Mitigations:
ProtonDB is right that I need to be on Proton 6.3-8 for it to work without launch options.

For higher versions, Launch Option "DXVK_FRAME_RATE=60 %command%" is not a fix.

**Comment by Bitwolfies** [comment]
So any version of experimental should already limit the FPS to 60 (I should know, im half the reason it does that) so you'd just be applying it twice on experimental.

You are right that it has broken now, but it's not due to the FPS cap.

**Comment by AlleyDismay** [comment]
> So any version of experimental should already limit the FPS to 60 (I should know, im half the reason it does that) so you'd just be applying it twice on experimental.
> 
> You are right that it has broken now, but it's not due to the FPS cap.

Yeah, my testing was wrong on the launch options. Corrected my OP.

**Comment by AJuujarvi** [comment]
@AlleyDismay Hi there! I've been trying to reproduce this and I am having troubles. I tested 5 different machines that had different hardware and software configurations and could not reproduce this issue. Your log also didn't seem to have anything in particular that would indicate an issue. 

As a shot in the dark, can you get a log with the launch options `PROTON_LOG=+quartz,+strmbase,+strmbase_qc,+gstreamer,+amstream,+qcap,+qcap_v4l,+qedit,+mciqtz,+mci,+mfplat,+evr,+dxva2,+wmvcore %command%` and attach it in a comment please.

**Comment by AlleyDismay** [comment]
> @AlleyDismay Hi there! I've been trying to reproduce this and I am having troubles. I tested 5 different machines that had different hardware and software configurations and could not reproduce this issue. Your log also didn't seem to have anything in particular that would indicate an issue.
> 
> As a shot in the dark, can you get a log with the launch options `PROTON_LOG=+quartz,+strmbase,+strmbase_qc,+gstreamer,+amstream,+qcap,+qcap_v4l,+qedit,+mciqtz,+mci,+mfplat,+evr,+dxva2,+wmvcore %command%` and attach it in a comment please.

@AJuujarvi - Here! Good luck! Gonna lose access to my testing computer for a week, so really hope you can replicate my issues, so we have a backup tester!

[steam-832320 - 8.0-3 - Tons of Options.log](https://github.com/ValveSoftware/Proton/files/12795950/steam-832320.-.8.0-3.-.Tons.of.Options.log)

**Comment by AJuujarvi** [comment]
@AlleyDismay Do you have a controller plugged in at all? Or some sort of adapter for a controller? What kind of keyboard and mouse do you have? We've had issues with specific peripherals as well in the past, and maybe that is what is happening here.

**Comment by AJuujarvi** [comment]
@AlleyDismay Oh I did find something interesting in your log just now at line 6023. `1343.242:01a4:0268:warn:seh:dispatch_exception EXCEPTION_ACCESS_VIOLATION exception (code=c0000005) raised`. Looking at it a little bit, I might want some input logs. Can you get me a new log running `PROTON_LOG=+hid,+hidp,+hid_report,+plugplay,+rawinput,+dinput,+xinput,+joystick,+setupapi,+input %command%` this time? Then can you get another log that is just `PROTON_LOG=+relay %command%` please? I think those could be very interesting here.

**Comment by AlleyDismay** [comment]
> @AlleyDismay Do you have a controller plugged in at all? Or some sort of adapter for a controller? What kind of keyboard and mouse do you have? We've had issues with specific peripherals as well in the past, and maybe that is what is happening here.

@AJuujarvi - Just disconnected my keyboard, and no problems. Putting it back in, and got a CTD immediately: `lsusb` reports:

`Bus 003 Device 006: ID 04b3:301b IBM Corp. SK-8815 Keyboard`

[steam-Keyboard Disconnect then re.log](https://github.com/ValveSoftware/Proton/files/12796558/steam-Keyboard.Disconnect.then.re.log)

Gonna do your other request now.

**Comment by AlleyDismay** [comment]
> @AlleyDismay Oh I did find something interesting in your log just now at line 6023. `1343.242:01a4:0268:warn:seh:dispatch_exception EXCEPTION_ACCESS_VIOLATION exception (code=c0000005) raised`. Looking at it a little bit, I might want some input logs. Can you get me a new log running `PROTON_LOG=+hid,+hidp,+hid_report,+plugplay,+rawinput,+dinput,+xinput,+joystick,+setupapi,+input %command%` this time? Then can you get another log that is just `PROTON_LOG=+relay %command%` please? I think those could be very interesting here.

@AJuujarvi - Seems like I do have to do keyboard inputs to get a crash. And getting into a game fast enough seems to also cure things?

[steam-832320 - 8.0-3 input options.log](https://github.com/ValveSoftware/Proton/files/12796588/steam-832320.-.8.0-3.input.options.log)

And the Relay log was more than 0.5 GB.

[steam-832320 - 8.0-3 relay option - froze instead of CTD.zip](https://github.com/ValveSoftware/Proton/files/12796601/steam-832320.-.8.0-3.relay.option.-.froze.instead.of.CTD.zip)

**Comment by AJuujarvi** [comment]
@AlleyDismay Thank you so much! I think we have what we need for now. If we need anything else, I will let you know. :)

**Comment by ivyl** [comment]
@AlleyDismay it seems to have nothing to do with the actual keyboard. The game, when there's no xbox controller plugged in, tried to load and unload xinput1_4.dll in a loop. You were just unlucky to hit a pre-existing race condition.

This should be fixed in [bleeding-edge beta branch of Proton Experimental](https://github.com/ValveSoftware/Proton/wiki/Proton-Versions#proton-bleeding-edge). Can you give it a spin? Hopefully this fixes your issue.

**Comment by AlleyDismay** [comment]
@ivyl - Seems like it did it! Thanks!

**Comment by AlleyDismay** [comment]
And like LunarLux, this game is also working on 8.05 and 9 (Beta), or at least I can navigate the screens without crashing, and that was the real issue.

Should this be closed and Regression tag removed now?
